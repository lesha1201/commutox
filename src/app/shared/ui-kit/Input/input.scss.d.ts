export const inputWrapper: string;
export const addon: string;
export const input: string;
export const small: string;
export const medium: string;
export const large: string;
export const full: string;
export const withAddons: string;
export const errorInput: string;
