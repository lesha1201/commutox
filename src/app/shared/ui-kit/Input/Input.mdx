---
name: Input
menu: UI Kit
---

import { Playground, PropsTable } from 'docz';
import Margin from 'src/app/shared/docz-helpers/Margin';
import Input from './index';

# Input

## Props

<PropsTable of={Input} />

## Default

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Input placeholder="First name" />
  </div>
</Playground>

## Width modifiers

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Margin type="bottom">
      <Input placeholder="Small" width="small" />
    </Margin>
    <Margin type="bottom">
      <Input placeholder="Medium" width="medium" />
    </Margin>
    <Margin type="bottom">
      <Input placeholder="Large" width="large" />
    </Margin>
    <Margin type="top">
      <Input placeholder="Full" width="full" />
    </Margin>
  </div>
</Playground>

## Before/after addons

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Margin type="bottom">
      <Input placeholder="Addons" addonBefore="Before" />
    </Margin>
    <Margin type="bottom">
      <Input placeholder="Addons" addonAfter="After" />
    </Margin>
    <Margin type="bottom">
      <Input placeholder="Addons" addonBefore="Before" addonAfter="After" />
    </Margin>
  </div>
</Playground>

## Disabled

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Input placeholder="Disabled" disabled />
  </div>
</Playground>

## Error

<Playground>
  <div style={{ fontSize: '14px' }}>
    <Input placeholder="Error" hasError />
  </div>
</Playground>
