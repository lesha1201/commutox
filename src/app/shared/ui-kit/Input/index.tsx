import cx from 'classnames';
import * as React from 'react';

import style from './input.scss';

/* Typings */
export interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
  width: 'small' | 'medium' | 'large' | 'full';
  className?: string;
  /** Adds an addon block before input */
  addonBefore?: string;
  /** Adds an addon block after input */
  addonAfter?: string;
  /** If it's set to true when input changes visually to show there is an error */
  hasError?: boolean;
  children?: never;
}

/* Component */
class Input extends React.Component<IProps, {}> {
  /* üóø Static properties ------------*/
  static defaultProps = {
    type: 'text',
    width: 'full',
  };
  /* End of Static properties -------*/

  /* ‚ôªÔ∏è Lifecycle -------------------*/
  render() {
    const { addonBefore, addonAfter } = this.props;

    if (addonBefore || addonAfter) {
      return this.getInputWithAddons();
    } else {
      return this.getInput();
    }
  }
  /* End of Lifecycle ---------------*/

  /* üîí Private methods -------------*/
  private getInput(hasAddons = false): React.ReactNode {
    const {
      width,
      className,
      addonBefore,
      addonAfter,
      hasError,
      ...domAttrs
    } = this.props;

    const cn = cx(
      hasError ? style.errorInput : style.input,
      hasAddons ? style.withAddons : style[width],
      className,
    );

    return <input className={cn} {...domAttrs} />;
  }

  private getInputWithAddons(): React.ReactNode {
    const { addonBefore, addonAfter } = this.props;
    const addonCn = cx(style.addon);

    return (
      <div className={style.inputWrapper}>
        {addonBefore && <span className={addonCn}>{addonBefore}</span>}
        {this.getInput(true)}
        {addonAfter && <span className={addonCn}>{addonAfter}</span>}
      </div>
    );
  }
  /* End of Private methods ---------*/
}

export default Input;
