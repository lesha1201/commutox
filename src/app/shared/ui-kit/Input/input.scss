@import 'src/app/shared/ui-kit/sass/variables';

$padding-top: 0.66em;
$padding-left: 0.83em;

.inputWrapper {
  display: flex;
  width: 100%;

  & > .addon,
  & > .input {
    border: 1px solid $color-gray-1;
    border-radius: 0;
  }

  & > *:first-child {
    border-top-left-radius: 0.33em;
    border-bottom-left-radius: 0.33em;
  }

  & > .addon:first-child {
    border-right-color: transparent;
  }

  & > *:last-child {
    border-top-right-radius: 0.33em;
    border-bottom-right-radius: 0.33em;
  }

  & > .addon:last-child {
    border-left-color: transparent;
  }
}

.input {
  box-sizing: border-box;
  padding: $padding-top $padding-left;

  border: 1px solid $color-gray-1;
  border-radius: 0.33em;
  color: inherit;
  outline: none;
  transition: all 0.2s;

  font-size: 1em;
  font-family: inherit;

  &:disabled {
    background: $color-light-gray-1;
    border-color: $color-gray-1;

    &::placeholder {
      color: $color-gray-1;
    }
  }

  &:focus {
    border: 1px solid $color-primary;
    color: $color-primary;
  }

  &.small {
    width: 130px;
  }

  &.medium {
    width: 200px;
  }

  &.large {
    width: 500px;
  }

  &.full {
    width: 100%;
  }

  &.withAddons {
    flex: 1;
  }
}

.error-input {
  composes: input;

  border-color: $color-danger;
  color: $color-danger;

  &:focus {
    border-color: $color-danger;
    color: $color-danger;
  }
}

.addon {
  padding: $padding-top $padding-left;
}
