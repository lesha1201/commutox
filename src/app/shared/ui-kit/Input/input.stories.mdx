import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';

import { Margin } from 'app/shared/storybook-helpers';
import Input from './index';

# Input

<Meta title="UI Kit/Input" component={Input} />

## Stories

### Default

<Preview withToolbar>
  <Story name="default">
    <div style={{ fontSize: '14px' }}>
      <Input placeholder="First name" />
    </div>
  </Story>
</Preview>

### Width modifiers

<Preview withToolbar>
  <Story name="width modifiers">
    <div style={{ fontSize: '14px' }}>
      <Margin type="bottom">
        <Input placeholder="Small" width="small" />
      </Margin>
      <Margin type="bottom">
        <Input placeholder="Medium" width="medium" />
      </Margin>
      <Margin type="bottom">
        <Input placeholder="Large" width="large" />
      </Margin>
      <Margin type="top">
        <Input placeholder="Full" width="full" />
      </Margin>
    </div>
  </Story>
</Preview>

### Before/after addons

<Preview withToolbar>
  <Story name="before/after addons">
    <div style={{ fontSize: '14px' }}>
      <Margin type="bottom">
        <Input placeholder="Addons" addonBefore="Before" />
      </Margin>
      <Margin type="bottom">
        <Input placeholder="Addons" addonAfter="After" />
      </Margin>
      <Margin type="bottom">
        <Input placeholder="Addons" addonBefore="Before" addonAfter="After" />
      </Margin>
    </div>
  </Story>
</Preview>

### Disabled

<Preview withToolbar>
  <Story name="sisabled">
    <div style={{ fontSize: '14px' }}>
      <Input placeholder="Disabled" disabled />
    </div>
  </Story>
</Preview>

### Error

<Preview withToolbar>
  <Story name="error">
    <div style={{ fontSize: '14px' }}>
      <Input placeholder="Error" hasError />
    </div>
  </Story>
</Preview>

## Props

<Props of={Input} />
