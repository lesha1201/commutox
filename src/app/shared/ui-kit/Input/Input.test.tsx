import * as React from 'react';
import { render } from 'react-testing-library';

import Input from '.';

describe('<Input />', () => {
  it('should render', () => {
    const { container } = render(<Input />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('with addons', () => {
    const { container } = render(<Input addonAfter="after" addonBefore="before" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
