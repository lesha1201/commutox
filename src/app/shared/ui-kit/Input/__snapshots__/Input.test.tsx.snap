// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render 1`] = `
<input
  class="input full"
  type="text"
/>
`;

exports[`<Input /> with addons 1`] = `
<div
  class="inputWrapper"
>
  <span
    class="addon"
  >
    before
  </span>
  <input
    class="input withAddons"
    type="text"
  />
  <span
    class="addon"
  >
    after
  </span>
</div>
`;
